### PR TITLE
Refresh access token - Add new method to use Esi internal refresh_token

### PR DESCRIPTION
**Pull request by normegil**
I've added a method to use the otherwise unused refresh_token available in the `Esi` struct. I found it a bit more convenient than a call that would look like this:

`if let Some(token) = esi.refresh_token {
     esi.use_refresh_token(token)?
}`

I've also change `use_refresh_token` to call the new method, to avoid duplicate code. I've kept `use_refresh_token` to not break the API.

### CODE REVIEW

File: src/jwt_util.rs [hunk 1]
```diff
@@ -43,10 +43,14 @@ async fn get_rs256_key(client: &Client) -> EsiResult<String> {
 }
 
 /// Decode and validate the JWT token
-fn validate(token: &str, client_id: &str, decoding_key: &DecodingKey) -> Result<TokenClaims, EsiError> {
+fn validate(token: &str, client_id: Option<&str>, decoding_key: &DecodingKey) -> Result<TokenClaims, EsiError> {
     let mut validations = Validation::new(Algorithm::RS256);
     validations.required_spec_claims = vec![String::from("sub")].into_iter().collect();
-    validations.set_audience(&[client_id, "EVE Online"]);
+    let mut aud = vec!["EVE Online"];
+    if let Some(id) = client_id {
```
Comment (Celeo) [inline src/jwt_util.rs]: Do you spot a use-case where the library would be validating a JWT but wouldn't have a client ID?

Comment (normegil) [reply src/jwt_util.rs]: Not really actually. A wrong assumption on my part. I will correct that.

Comment (Celeo) [reply src/jwt_util.rs]: All good!

**Review by Celeo — APPROVED** [review]
Love it!

**Comment by Celeo** [comment]
Released in `0.27.0`.

**Pull request merged**